ibis/backends/base/__init__.py

```diff
@@ -593,3 +595,28 @@ def has_operation(cls, operation: type[ops.Value]) -> bool:
         raise NotImplementedError(
             f"{cls.name} backend has not implemented `has_operation` API"
         )
+
+
+def connect(url_or_path: str) -> BaseBackend:
+    'Connect to given *url_or_path*. Auto-detect backend from url scheme or path extension.'
+    path = Path(url_or_path)
```

Comment:
Can we reuse [`sqlalchemy.engine.make_url`](https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=make_url#sqlalchemy.engine.make_url) here?

The [`sqlalchemy.engine.URL`](https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=make_url#sqlalchemy.engine.URL) object nicely captures most or all of the things we care about and `make_url` handles the parsing includes the driver with `+`.

Reply:
Sure thing, I will check it out!

Reply:
> It seems like we still have to split the driver out ourselves

I think if you reduce the scope to not include the extension then it becomes:

```
In [18]: import ibis, sqlalchemy as sa

In [19]: url = sa.engine.url.make_url('duckdb:///ci/ibis-testing-data/ibis_testing.ddb')

In [20]: backend = getattr(ibis, url.drivername)

In [21]: con = backend.connect(url.database)
```

Reply:
What's the purpose of the `+`? Apologies if we discussed this already, I can't remember.

Reply:
I'm thinking we just start with `backend://` URLs for now, instead of supporting `backend://`, `ibis+backend://`, files with extensions matching the backend name.